sycl/doc/extensions/proposed/sycl_ext_oneapi_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,326 @@
+
+        range_type get_cluster_group_range() const;
+
+        id_type get_group_id() const;
+    }
+----
+
+
+[%header,cols="5,5"]
+|===
+|Method
+|Description
+
+|`linear_id get_group_linear_id() const`
+|Returns the flattened id of the calling `group` within the cluster
+
+|`linear_id get_local_linear_id() const`
+|Returns the flattened index of the calling work-item within the cluster
+
+|`range_type get_cluster_group_range() const`
+|Returns the number of work-groups in each dimension
+
+|`id_type get_group_id() const`
+|Returns the id of the calling work-group along each dimension
+|===
+
+
+To obtain the total number of clusters in the kernel, and to obtain the 
+id of the cluster of the calling work-item, this extension proposes to add two
+new member functions the `nd_item` class, namely 
+`ext_oneapi_cuda_get_size` and `ext_oneapi_cuda_get_cluster_id`
+
+
+[%header,cols="10,5"]
+|===
+
+|`range<3> nd_item::ext_oneapi_cuda_get_size(size_t) const`
+|Returns the total number of `cluster_groups` across each dimension.
+
+|`id<3> nd_item::get_cluster_id() const`
+|Returns the id of the cluster along each dimension.
+|===
+
+
+== Accessing another work-group's local memory
+
+Work-group's within the cluster have the ability to access another work-group's 
+local memory. Typically addresses which reside in the local memory of a 
+work-group can only be addressed by the work-items of that work-group. 
+Therefore, to access another work-group's local memory, the address needs to be
+mapped such that the address in another work-group is addressable within the 
+calling work-item. Further, to access another another work-group's local memory,
+all the work-groups within the cluster must exist and the work-group's should
+not cease to exist before all the memory operations are completed. This can be 
+ensured by synchronizing all the work-items within the cluster before and after
+the local memory operations.
+
+A member function of the `cluster_group` class; 
+`map_cluster_local_pointer` will perform the mapping and return a pointer 
+which can then be dereferenced by the calling work-item. 
+
+
+[%header,cols="10,5"]
+|===
+
+|T*  map_cluster_local_pointer(T* addr, size_t group_id)
+|Accepts the equivalent address to the memory location relative to the calling 
+work-item which is to be mapped from the local memory of the work-group, as 
+specified by the group_id, denoting the flattened group_id within the cluster
+|===
+
+
+== Cluster Memory Fence Scope and Barrier
+
+Work-items in a work-group can atomically operate on the local memory addresses
+of another work-group which have mapped as described above. To facilitate this,
+this proposal introduces a new memory scope, 
+`sycl_ext_oneapi_experimental_cuda_cluster_fence` which can be used with the
+existing `atomic_ref` class.
+
+[source,c++]
+----
+    class cluster_group {
+    public:
+    ...
+    static constexpr memory_scope fence_scope = 
```

Comment:
It looks like `memory_scope::sycl_ext_oneapi_experimental_cuda_cluster_fence` is used but there's no definition, we should define a new member of the `memory_scope` enum class:

```
enum class memory_scope {
...
sycl_ext_oneapi_experimental_cuda_cluster_group,
...
};

inline constexpr auto sycl_ext_oneapi_experimental_cuda_memory_scope_cluster_group
  = memory_scope::sycl_ext_oneapi_experimental_cuda_cluster_group;
```

And then can we define the value of `fence_scope` in the `cluster_group` class to this enum value.